sdk/core/core-rest-pipeline/src/policies/tracingPolicy.ts

```diff
@@ -71,89 +67,92 @@ export function tracingPolicy(options: TracingPolicyOptions = {}): PipelinePolic
-      ...(request.tracingOptions as any)?.spanOptions,
-      kind: SpanKind.CLIENT,
-    };
+    return createTracingClient({
+      namespace: "",
+      packageName: "@azure/core-rest-pipeline",
+      packageVersion: SDK_VERSION,
+    });
+  } catch (e: unknown) {
+    return undefined;
+  }
+}
 
-    // Passing spanOptions as part of tracingOptions to maintain compatibility @azure/core-tracing@preview.13 and earlier.
-    // We can pass this as a separate parameter once we upgrade to the latest core-tracing.
+function tryCreateSpan(
+  tracingClient: TracingClient,
+  request: PipelineRequest,
+  userAgent?: string
+): { span: TracingSpan; tracingContext: TracingContext } | undefined {
+  try {
     // As per spec, we do not need to differentiate between HTTP and HTTPS in span name.
-    const { span } = createSpan(`HTTP ${request.method}`, {
-      tracingOptions: { ...request.tracingOptions, spanOptions: createSpanOptions },
-    });
+    const { span, updatedOptions } = tracingClient.startSpan(
+      `HTTP ${request.method}`,
+      { tracingOptions: request.tracingOptions },
+      {
+        spanKind: "client",
+        spanAttributes: {
+          "http.method": request.method,
+          "http.url": request.url,
+          requestId: request.requestId,
+        },
+      }
+    );
 
     // If the span is not recording, don't do any more work.
     if (!span.isRecording()) {
       span.end();
       return undefined;
     }
 
-    const namespaceFromContext = request.tracingOptions?.tracingContext?.getValue(
-      Symbol.for("az.namespace")
-    );
-
-    if (typeof namespaceFromContext === "string") {
-      span.setAttribute("az.namespace", namespaceFromContext);
-    }
-
-    span.setAttributes({
-      "http.method": request.method,
-      "http.url": request.url,
-      requestId: request.requestId,
-    });
-
     if (userAgent) {
       span.setAttribute("http.user_agent", userAgent);
     }
 
     // set headers
-    const spanContext = span.spanContext();
-    const traceParentHeader = getTraceParentHeader(spanContext);
-    if (traceParentHeader && isSpanContextValid(spanContext)) {
-      request.headers.set("traceparent", traceParentHeader);
-      const traceState = spanContext.traceState && spanContext.traceState.serialize();
-      // if tracestate is set, traceparent MUST be set, so only set tracestate after traceparent
-      if (traceState) {
-        request.headers.set("tracestate", traceState);
-      }
+    const headers = tracingClient.createRequestHeaders(
+      updatedOptions.tracingOptions.tracingContext
+    );
+    for (const [key, value] of Object.entries(headers)) {
+      request.headers.set(key, value);
     }
-    return span;
-  } catch (error) {
-    logger.warning(`Skipping creating a tracing span due to an error: ${error.message}`);
+    return { span, tracingContext: updatedOptions.tracingOptions.tracingContext };
+  } catch (e) {
+    const message = isError(e) ? e.message : "";
```

Comment:
Curious about this - is there a scenario where we catch null or undefined? Even if it's not ErrorLike e.message should still yield undefined which gets interpolated to "" below but maybe I am thinking about this wrong? 

The only time I think this will be problematic is if someone `throw undefined` or something because even primitives should respond to `.<property>`

```ts
> y = 5; y.message
undefined
> y = "foo"; y.message
undefined
> y = true; y.message
undefined
```

Reply:
I'm open to how we should fallback here, maybe we could provide a helper that stringified whatever we got if it's not an Error object? something like

```ts
function getMessage(e: unknown): string {
   if (isError(e)) { 
      return e.message 
   } else {
      return `Unknown error ${String(e)}`;
   }
}
```